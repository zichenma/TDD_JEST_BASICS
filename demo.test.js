import {
  generateConfig,
  generateAnotherConfig,
  generateThirdConfig
} from "./demo";

// test('test generateConfig', () => {
//     // expect(generateConfig()).toEqual({
//     //     server: 'http://localhost',
//     //     port: 8080,
//     //     domain: 'localhost'
//     // })
//     expect(generateConfig()).toMatchSnapshot();
// });

// test('test generateAnotherConfig', () => {
//     expect(generateAnotherConfig()).toMatchSnapshot();
// });

test.('test generateThirdConfi', () => {
    expect(generateThirdConfig()).toMatchSnapshot({
        time : expect.any(Date) // Eg，Number, String....
    });
});

// 需要安装： npm install prettier@1.18.2 再用 toMatchInlineSnapshot: 
test.only("test generateThirdConfi", () => {
  expect(generateThirdConfig()).toMatchInlineSnapshot(
    {
      time: expect.any(Date)
    },
    `
    Object {
      "domain": "localhost",
      "port": 8080,
      "server": "http://localhost",
      "time": Any<Date>,
    }
  `
  );
});
