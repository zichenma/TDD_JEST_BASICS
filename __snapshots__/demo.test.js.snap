// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test generateThirdConfi 1`] = `
Object {
  "domain": "localhost",
  "port": 8080,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;
